modules/ossm-threescale-webassembly-module-examples-for-credentials-use-cases.adoc

```diff
@@ -0,0 +1,236 @@
+
+You can combine them all, although when you specify multiple source objects with their own `lookup queries`, they are evaluated in order until one of them successfully resolves.
+
+[id="api-key-in-query-string-parameters_{context}"]
+== API key (`user_key`) in query string parameters
+This looks up a `user_key` in a query string parameter or header of the same name:
+
+[source,yaml]
+----
+credentials:
+  user_key:
+    - query_string:
+        keys:
+          - user_key
+    - header:
+        keys:
+          - user_key
+----
+
+[id="application-id-and-key_{context}"]
+== Application ID and `key`
+The following example looks up a `app_key` and `app_id` in a query or headers.
+
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - app_id
+    - query_string:
+        keys:
+          - app_id
+  app_key:
+    - header:
+        keys:
+          - app_key
+    - query_string:
+        keys:
+          - app_key
+----
+
+[id="authorization-header_{context}"]
+== Authorization header
+A request includes an `app_id` and `key` in an `authorization` header. If there is at least one or two values outputted at the end, then you can assign the `app_key`.
+
+The resolution here will assign the `application key` if there is one or two output at the end.
+
+The `authorization` header specifies a value with the type of authorization and its value is encoded as `Base64`. This means you can split the value by a space character, take the second output and then split it again using a colon (:) as the separator. For example, if you use this format: `app_id:app_key`, the header will look like this for `credential`:
+
+`aladdin:opensesame`:  Authorization: Basic YWxhZGRpbjpvcGVuc2VzYW1l
+
+You must use lower case header field names as shown in the following example:
+
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - authorization
+        ops:
+          - split:
+              separator: " "
+              max: 2
+          - length:
+              min: 2
+          - drop:
+              head: 1
+          - base64_urlsafe
+          - split:
+              max: 2
+  app_key:
+    - header:
+        keys:
+          - app_key
+----
+
+The previous example use case looks at the headers for an `authorization`:
+
+. It takes its string value and split it by a space, checking that it generates at least two values of a `credential`-type and the `credential` itself, then dropping the `credential`-type.
+. It then decodes the second value containing the data it needs, and splits it by using a colon (:) character to have an operations stack including first the `app_id`, then the `app_key`, if it exists.
+.. If `app_key` does not exist in the authorization header then its specific sources are checked, for example, the header with the key `app_key` in this case.
++
+You can always add extra conditions to `credentials`, for example:
```

Comment:
This "for example" is currently at the step 2a level, and then jumps back up a level to a step 3. Were the following bullets supposed to be indented under 2a, or is this intended?

Reply:
Thanks...fixed :)